### PR TITLE
VACMS-19516 Fix email input focus on error

### PR DESCRIPTION
**Pull request by randimays**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
When the email signup widget has an error, the focus should be thrown to the header of the input so that the next element in the tab order is the input. This fixes that issue as well as resolves an issue with not enough color contrast between the character counter text and the blue background. ([Color contrast guidance from Laura here](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19516#issuecomment-2484157450)).

Also includes a tweak to make the input a `VaTextInput` so it will be properly attached to the React binding and event listeners. It was causing a console error that I did not notice before.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19516

## Testing done
**Requires testing locally with content-build running and pointed to https://cms-g1rglcuobmlvgewnhny9uo7f6vigntwi.demo.cms.va.gov/** - This is how you get the new email form because the feature toggle is in the correct state. Otherwise you will only see the old form.

1. Turn on Mac VoiceOver so you can tell where focus is thrown
2. Type in an invalid email address and tab out of the field (blur the input)
3. Validate that the focus is thrown to the header of the input (`Sign up to get the latest VA updates`)
4. Immediately after the header is read, the error message should be read
5. If you hit the tab key, the next element that you should focus on is the input where you can edit the email
6. Edit the email address to a valid input
7. Tab again to Sign up and verify that the button works correctly

## Screenshots

For the focus updates:

https://github.com/user-attachments/assets/73ae9c8b-cae9-418e-8a17-4c269474ad64

For the color contrast updates:

Screenshot 2024-11-19 at 10 15 18 AM

Screenshot 2024-11-19 at 10 15 13 AM

## Acceptance criteria

- [x] Focus should be thrown to the header of the component so the next tabbable element is the input

### CODE REVIEW

**Comment by randimays** [comment]
@laflannery If you would like to review the changes for focus behavior and character count color, I dropped a video and some screenshots here. The code changes require running both VW and CB together locally and I don't think you want to do (though - let me know if you're ever interested).

The review instance won't work as expected because it requires a CMS feature toggle being in the correct state and I don't think I can reliably change that without affecting other PRs.

I will let you know when this code merges and is ready on the other Tugboat you validated before so you can physically test it. Note that there wasn't a PR for the prior changes because I had to get everything merged to `main` and **then** set up the Tugboat in order to get all the things in the correct place.

**Comment by laflannery** [comment]
I watched the video, that looks good. Thanks!

**Comment by chriskim2311** [comment]
@randimays Hmm I am not getting the error message read to me after tabbing to the header after the invalid email address.

**Comment by randimays** [comment]
Spoke with Chris and Laura in DMs about the focus behavior being different than expected on Chris's machine. We think it has to do with different macOS versions (and thus different Mac VoiceOver behavior) and the way the web components are laid out in markup.

Laura said accessibility folks have been looking at error handling in general on form fields and there are known issues with error messages being read out as expected. We're going to accept the behavior differences as-is and work on getting this addressed at the web component level (most likely).

**Review by chriskim2311 — APPROVED** [review]
LGTM!

File: src/applications/static-pages/homepage-email-signup/EmailSignup.jsx [hunk 1]
```diff
@@ -1,10 +1,53 @@
...
+        }
+
+        if (inputHeader && inputError) {
+          inputHeader.setAttribute('tabindex', '-1');
+          inputHeader.focus();
+          setHeaderHasFocused(true);
+        }
+      }
+    },
+    [inputError],
```
Comment (harshil1793) [inline src/applications/static-pages/homepage-email-signup/EmailSignup.jsx:49]: It looks like the new code at line 49 introduces a new linting warning. Can we please clean these up before we merge this?

Comment (randimays) [reply src/applications/static-pages/homepage-email-signup/EmailSignup.jsx:49]: @harshil1793 Logically it does not make sense to add the dependency it's suggesting, so I didn't add it.

**Review by harshilmoradia — APPROVED** [review]

**Pull request merged**